### ISSUE TITLE
🐞 Clicking max at voting and farming trims the amount, and rounds up the value, resulting in an error

### ISSUE THREAD

**Issue by mwaddip** [b-u-i-d-l/dfo-hub#32]
# **Describe the bug**

When adding liquidity for farming, or adding votes to a proposal, clicking the max button will sometimes result in an error saying that you don't have enough balance. This is due to the fact that the value is trimmed to a certain amount of decimals, and then rounded up.

# **To Reproduce**

1. Have (for example) 8.88888888888888888 of a token.
2. Go to either 'Governance - Proposals - Vote' or 'DeFi - Farming - Stake', and click the max button.
3. Value will show something like 8.88889 
4. Submit
5. Get error that you don't have enough balance to do so.

# **Expected behavior**

Basically in step 3, the value has to either be trimmed while rounding down by default, or the full value with all its decimals should be used.
Expected result in step 3: Value will show 8.88888

# **Web3 Scenario (please complete the following information):**

 - Device: Any
 - OS: Any
 - Browser: Any
 - Version Any
 - Provider N/A

**Comment by alet89** [comment]
Solved
